File: files/CH-216.py
import pandas as pd
import re

path = "CH-216 Column Splitting.xlsx"

input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D:F", skiprows=1, nrows=6)
result = input["ID"].str.extract(r"^([A-Z]{2})(.*)([A-Za-z0-9]{1})$")
result.columns = ["Prefix", "Root", "Suffix"]

File: files/CH-216.R
library(tidyverse)
library(readxl)

path = "files/CH-216 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:F7")

result = input %>%
  extract(ID, into = c("Prefix", "Root", "Suffix"),
          regex = "^([A-Z]{2})(.*)([A-Za-z0-9]{1})$")

all.equal(result, test)
#> [1] TRUE